scipy/sparse/csgraph/_flow.pyx

```diff
@@ -0,0 +1,398 @@
+    .. [2] Cormen, T. H. and Leiserson, C. E. and Rivest, R. L. and Stein C.
+           Introduction to Algorithms. Second Edition. 2001. MIT Press.
+
+    Examples
+    --------
+    Perhaps the simplest flow problem is that of a graph of only two vertices
+    with an edge from source (0) to sink (1)::
+
+        (0) --5--> (1)
+
+    Here, the maximum flow is simply the capacity of the edge:
+
+    >>> from scipy.sparse import csr_matrix
+    >>> from scipy.sparse.csgraph import maximum_flow
+    >>> graph = csr_matrix([[0, 5], [0, 0]])
+    >>> maximum_flow(graph, 0, 1).flow_value
+    5
+
+    If, on the other hand, there is a bottleneck between source and sink, that
+    can reduce the maximum flow::
+
+        (0) --5--> (1) --3--> (2)
+
+    >>> graph = csr_matrix([[0, 5, 0], [0, 0, 3], [0, 0, 0]])
+    >>> maximum_flow(graph, 0, 2).flow_value
+    3
+
+    A less trivial example is given in [2]_, Chapter 26.1:
+
+    >>> graph = csr_matrix([[0, 16, 13,  0,  0,  0],
+    ...                     [0, 10,  0, 12,  0,  0],
+    ...                     [0,  4,  0,  0, 14,  0],
+    ...                     [0,  0,  9,  0,  0, 20],
+    ...                     [0,  0,  0,  7,  0,  4],
+    ...                     [0,  0,  0,  0,  0,  0]])
+    >>> maximum_flow(graph, 0, 5).flow_value
+    23
+
+    It is possible to reduce the problem of finding a maximum matching in a
+    bipartite graph to a maximum flow problem: Let :math:`G = ((U, V), E)` be a
+    bipartite graph. Then, add to the graph a source vertex with edges to every
+    vertex in :math:`U` and a sink vertex with edges from every vertex in
+    :math:`V`. Finally, give every edge in the resulting graph a capacity of 1.
+    Then, a maximum flow in the new graph gives a maximum matching in the
+    original graph consisting of the edges in :math:`E` whose flow is positive.
+
+    Assume that the edges are represented by a
+    :math:`\lvert U \rvert \times \lvert V \rvert` matrix in CSR format whose
+    :math:`(i, j)`'th entry is 1 if there is an edge from :math:`i \in U` to
+    :math:`j \in V` and 0 otherwise; that is, the input is of the form required
+    by :func:`maximum_bipartite_matching`. Then the CSR representation of the
+    graph constructed above contains this matrix as a block. Here's an example:
+
+    >>> graph = csr_matrix([[0, 1, 0, 1], [1, 0, 1, 0], [0, 1, 1, 0]])
+    >>> print(graph.toarray())
+    [[0 1 0 1]
+     [1 0 1 0]
+     [0 1 1 0]]
+    >>> i, j = graph.shape
+    >>> n = graph.nnz
+    >>> indptr = np.concatenate([[0],
+    ...                          graph.indptr + i,
+    ...                          np.arange(n + i + 1, n + i + j + 1),
+    ...                          [n + i + j]])
+    >>> indices = np.concatenate([np.arange(1, i + 1),
+    ...                           graph.indices + i + 1,
+    ...                           np.repeat(i + j + 1, j)])
+    >>> data = np.ones(n + i + j, dtype=np.int)
```

Comment:
Note:
```
>>> np.int is int                                                                              
True
```

You want to be using `np.int32`, `np.int64`, or the Python builtin `int`. Those aliases are in the numpy namespace for historical reasons and are best avoided